src/ocean/io/compress/ZipStream.d

```diff
@@ -0,0 +1,492 @@
+/*******************************************************************************
+
+    Zlib decoder which supports pkzip and gzip archives, and can be stored in a
+    pool.
+
+    In general it is not possible to stream pkzip archives. This is because the
+    format supports some quirky features which were important in the days of
+    floppy diskettes.
+
+    This class supports an important case where streaming is
+    possible: an archive which consists of a single file stored at
+    the start of the archive, using DEFLATE compression.
+
+    Needs linking with -lz.
+
+    Usage example:
+
+    ---
+
+        import ocean.io.compress.ZipStream;
```

Comment:
Normally I would be a fan of having example code as unit tests. Not sure if this is possible for this module though.

Reply:
Even if you don't run the code, wrapping the code inside a function that is never executed is already a step up because it ensures it'll always compile.

Reply:
It's not real code though, it is pseudocode, It uses a hypothetical function, which would presumably be called from a separate fiber. So it will not compile.
It's a modified version of the example in `ZlibStreamDecompressor`.

Reply:
So there is no way we could just replace it with a dummy function?

Reply:
I think that would clutter the example so much it would be largely useless as documentation. The point of it is to tell the higher-level reasoning behind using the code. Ie it is the WHY, not the WHAT.
The unittest which I have added, covers what's actually required to use the class.